src/components/ProductGridComponent.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import '../Styles/grid.css'
+import { GridComponent } from './GridComponent';
+
+export const ProductGridComponent = ({json}) => {
+    let grid=[];
+    json.results.forEach(product=>{
```

Comment:
Here you can have a performance leak, be careful when using local variables instead of component state, as React handle it different across renders, specially if you have parent / children components